src/librustc_typeck/collect.rs

```diff
@@ -1918,7 +1918,10 @@ fn explicit_predicates_of<'a, 'tcx>(
         }
     }
 
-    let hir_id = tcx.hir().as_local_hir_id(def_id).unwrap();
+    let hir_id = match tcx.hir().as_local_hir_id(def_id) {
+        Some(hir_id) => hir_id,
+        None => return tcx.predicates_of(def_id),
```

Comment:
This is another change to `rustc_typeck` that wasn't explicitly approved (the comment thread doesn't make it clear that *a hack* was added outside of rustdoc), can we please not do this? I'll create a PR reverting these, but I feel like we have a larger process break issue...

Yes, I didn't see this notification until now, but *please* ping me elsewhere instead of merging typeck hacks like this.

Reply:
Not to mention this is wrong for cross-crate items, my expectation was that `explicit_predicates_of` was available cross-crate.

Reply:
@estebank ^

Reply:
I did ping the compiler team so I thought you'd see the notification, my bad.

Reply:
Talked with Eddy on chat. My bad as the previous seemed to make sense on isolation. I'll defer to Eddy's assessment on typeck code in the future.

Reply:
@GuillaumeGomez Being notified isn't enough, we need to acknowledge it too.
With @nikomatsakis less active lately (and also on PTO more recently), a bit more care should be taken.

Reply:
Opened #59789.

Reply:
I didn't approved the PR nor asked anyone (except the compiler team) to review it. I approve for more care to be taken but I feel like taking the blame for an error I didn't make. :-/

Reply:
This is a process failure, not your failure! We'll figure out how to prevent such things happening in the future.

Reply:
@GuillaumeGomez The one at fault here was _me_, not you, as I approved this change thinking it looked reasonable without being my main area of expertise, _I_ should have deferred to @eddyb's or @nikomatsakis' judgement on this case.